src/query.jl

```diff
@@ -366,6 +366,7 @@ format inferred from the file's extension and/or magic bytes.
 """
 function query(filename::AbstractString)
     _, ext = splitext(filename)
+    filename = abspath(filename)
```

Comment:
I guess it's a question of when we want to "immortalize" the path. I was deliberate in my choices, thinking about this kind of chicanery:

```jl
tim@diva:~$ cd ~
tim@diva:~$ julia -q
julia> using FileIO, Colors

julia> f = query("photo.png")  # here I just want the format, but not the path
FileIO.File{FileIO.DataFormat{:PNG}}("photo.png")

julia> cd(mktempdir())

julia> open(f, "w") do io
           save(io, rand(RGB{U8}, 5, 5))
       end

<photo.png is in the tempdir, not my homedir>
```

I was thinking this is a good thing.

But perhaps it's reasonable to postulate that the moment of calling `query` constitutes the best time to immortalize. We should choose whatever is "least surprising" and then document the behavior. If you feel strongly that this version is less surprising, I say let's go with it.